secret_store/src/listener/service_contract_listener.rs

```diff
@@ -301,16 +319,21 @@ impl Drop for ServiceContractListener {
 
 impl ChainNotify for ServiceContractListener {
 	fn new_blocks(&self, _imported: Vec<H256>, _invalid: Vec<H256>, enacted: Vec<H256>, _retracted: Vec<H256>, _sealed: Vec<H256>, _proposed: Vec<Bytes>, _duration: u64) {
-		self.data.contract.update();
-		if !self.data.contract.is_actual() {
+		let enacted_len = enacted.len();
+		if enacted_len == 0 {
```

Comment:
`enacted.is_empty()`?
EDIT: just noticed you are using `enacted_len` later on, so I guess it's fine.